py-polars/polars/series/series.py

```diff
@@ -426,15 +427,49 @@ def _from_buffer(
         dtype
             The data type of the buffer.
         buffer_info
-            Tuple containing buffer information in the form (pointer, offset, length).
-        base
+            Tuple containing buffer information in the form `(pointer, offset, length)`.
+        owner
             The object owning the buffer.
 
         Returns
         -------
         Series
         """
-        return self._from_pyseries(PySeries._from_buffer(dtype, buffer_info, base))
+        return self._from_pyseries(PySeries._from_buffer(dtype, buffer_info, owner))
+
+    @classmethod
+    def _from_buffers(
+        self,
+        dtype: PolarsDataType,
+        data: Series,
+        validity: Series | None = None,
+        offsets: Series | None = None,
```

Comment:
This will change with the new string type. I think we should accept a validity and a list of buffers. Then the according datatype will know how to handle them.

Reply:
How will it change, exactly? Offsets will be replaced by the new u128 buffer, right? Then we can simply rename the offsets parameter - it's a private method.

If we add additional buffer types later, I think it'll be more clear to have those as additional optional parameters, rather than have a list of buffers where the user has to somehow know the right ordering.

Reply:
All right, I updated the function signature!